src/main/java/com/hubspot/smtp/messages/MessageContent.java

```diff
@@ -0,0 +1,27 @@
+package com.hubspot.smtp.messages;
+
+import java.io.InputStream;
+
+import io.netty.buffer.ByteBuf;
+
+public abstract class MessageContent {
+  public static MessageContent of(ByteBuf messageBuffer) {
+    return new ByteBufMessageContent(messageBuffer, true);
+  }
+
+  public static MessageContent of(InputStream messageStream, int size, boolean applyDotStuffing) {
```

Comment:
nit: I would rather just create an enum rather than a boolean here. It becomes more readable in the calling code. E.g.:

```java
MessageContent.of(myInputStream, 2048, false);
```

becomes:

```java
MessageContent.of(myInputStream, 2048, MessageContentEncoding.ASSUME_DOT_STUFFED);
```